### PR TITLE
Add support for animating `startPoint` and `endPoint` of radial gradients

### LINKED ISSUE #1762: Add support for `animating multiple keyframes for gradient startPoint` with the Core Animation renderer
Hello! I was super excited to see the performance improvements in https://github.com/airbnb/lottie-ios/discussions/1627. Although switching to the newer version of the library and forcing the use of the Core Animation did yield good CPU usage results, there is an animation we use in our project that doesn't render properly.

These are the logs I get in the console when using Core Animation:

```
Lottie/AnimationView.swift:1046: Assertion failed: Encountered Core Animation compatibility issues while setting up animation:
[shimmering orb rotating -withBlueInFront 4.gradient- vertical flowdown.Gradient Fill 1] The Core Animation rendering engine does not support animating multiple keyframes for gradient startPoint values (due to limitations of Core Animation `CAKeyframeAnimation`s).
[shimmering orb rotating -withBlueInFront 4.mask-flowdown dup.Gradient Fill 1] The Core Animation rendering engine does not support animating multiple keyframes for gradient startPoint values (due to limitations of Core Animation `CAKeyframeAnimation`s).
```

## Which Version of Lottie are you using?

Lottie 3.4.4

## Expected Behavior 

https://user-images.githubusercontent.com/13305965/194884059-083fce3b-b18e-4874-9f38-9dfd3fda4906.mov

## Actual Behavior

https://user-images.githubusercontent.com/13305965/194884191-90202011-399e-4bfb-ad8f-8fbb6a40cfb9.mov

## Animation JSON

[shield-button.json.zip](https://github.com/airbnb/lottie-ios/files/9746573/shield-button.json.zip)

### PR DESCRIPTION
**Pull request by calda**
This PR adds support for animating the `startPoint` and `endPoint` of radial gradients. We can do this by combining the two sets of keyframes into a single `KeyframeGroup`, manually interpolating them if necessary. Fixes #1762. 

I took this opportunity to consolidate the manual interpolation functionality into `Keyframes.combined`, and also adopt this in most other places where we were using `KeyframeGroup.exactlyOneKeyframe`.

| Before | After |
| ---- | ---- |
|  |  |

### CODE REVIEW

File: Sources/Private/CoreAnimation/Extensions/Keyframes+combined.swift [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+// MARK: - Keyframes
+
+enum Keyframes {
+
+  // MARK: Internal
+
+  /// Combines the given keyframe groups of `Keyframe<T>`s into a single keyframe group of of `Keyframe<[T]>`s
+  ///  - If all of the `KeyframeGroup`s have the exact same animation timing, the keyframes are merged
+  ///  - Otherwise, the keyframes are manually interpolated at each frame in the animation
+  static func combined<T>(_ allGroups: [KeyframeGroup<T>]) -> KeyframeGroup<[T]>
```
Comment (calda) [inline Sources/Private/CoreAnimation/Extensions/Keyframes+combined.swift:13]: These methods were renamed from `Keyframes.combinedIfPossible(...) -> KeyframeGroup?` to `Keyframes.combined(...) -> KeyframeGroup`

File: Sources/Private/CoreAnimation/Extensions/Keyframes+combined.swift [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    // to check those.
+    let animatingKeyframes = untypedGroups.filter { $0.keyframes.count > 1 }
+
+    guard
+      !allGroups.isEmpty,
+      animatingKeyframes.allSatisfy({ $0.hasSameTimingParameters(as: animatingKeyframes[0]) })
+    else {
+      // If the keyframes don't all share the same timing information,
+      // we have to interpolate the value at each individual frame
+      return Keyframes.manuallyInterpolated(allGroups, makeCombinedResult: makeCombinedResult)
```
Comment (calda) [inline Sources/Private/CoreAnimation/Extensions/Keyframes+combined.swift:126]: This is the only new logic in this method (previously this was just `return nil`)

File: Sources/Private/CoreAnimation/Extensions/Keyframes+combined.swift [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        LottieLogger.shared.assertionFailure("""
+          Failed to cast untyped keyframe values to expected type. This is an internal error.
+          """)
+      }
+    }
+
+    return KeyframeGroup(keyframes: combinedKeyframes)
+  }
+
+  private static func manuallyInterpolated<CombinedResult>(
```
Comment (calda) [inline Sources/Private/CoreAnimation/Extensions/Keyframes+combined.swift:148]: This method is new

File: Sources/Private/CoreAnimation/Extensions/Keyframes+combined.swift [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        else { return nil }
+
+        return makeCombinedResult(t1, t2, t3)
+      })
+  }
+
+  /// Combines the given keyframe groups of `Keyframe<T>`s into a single keyframe group of of `Keyframe<[T]>`s
+  ///  - If all of the `KeyframeGroup`s have the exact same animation timing, the keyframes are merged
+  ///  - Otherwise, the keyframes are manually interpolated at each frame in the animation
+  static func combined<T1, T2, T3, T4, T5, T6, T7, CombinedResult>(
```
Comment (erichoracek) [inline Sources/Private/CoreAnimation/Extensions/Keyframes+combined.swift:70]: Could really use "row polymorphism" for this one! 😂

**Review by erichoracek — APPROVED** [review]
Looks great! Nice cleanup too

File: Sources/Private/CoreAnimation/Extensions/Keyframes+combined.swift (hunk 4 above)
Comment (calda) [reply Sources/Private/CoreAnimation/Extensions/Keyframes+combined.swift:70]: please, variadic generics when

**Pull request merged**